.github/scripts/parse_test_report.py

```
@@ -0,0 +1,198 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Parse JUnit XML test report, generate a markdown summary, and optionally post it to a PR."""
+
+import os
+import subprocess
+import xml.etree.ElementTree as ET
+
+
+def parse_report(xml_path: str, branches: dict[str, str], run_url: str) -> str:
```

Ruff (0.15.5)

[warning] Too many branches (13 > 12)

(PLR0912)